### PR TITLE
[All] (convert Visual Studio project to TypeScript) Updated instructions

### LINKED ISSUE #1142: Two things the article doesn't mention
For this to work properly I had to add the following ...
1. Debug didn't work. It complained about missing symbols. I added the sourcemap line to the tsconfig.json file. 

    ```
    {
      "compilerOptions": {
        "skipLibCheck": true,
        "lib": [ "es5", "dom", "es2015.promise" ],
        "sourceMap": true
      }
    }
    ```

2. After converting to .ts, all the Microsoft reference documentation change the calls to something like this ...

    ```
    // Run a batch operation against the Excel object model
    await Excel.run(async  (ctx) => {
        // Create a proxy object for the active sheet
        const sheet = ctx.workbook.worksheets.getActiveWorksheet();
    
        // Queue a command to write the sample data to the worksheet
        sheet.getRange("B3:D5").values = values;

        // Run the queued-up commands, and return a promise to indicate task completion
        await ctx.sync();
    })
    ```

This threw Javascript errors 'promise' is undefined. I found this link ... https://stackoverflow.com/questions/44461312/office-addins-file-in-its-typescript-version-doesnt-work

which fixed it, note that you have to scroll all the way to the bottom as the solution wasn't posted correctly at first. The short version is to add the line "(window as any).Promise = OfficeExtension.Promise;" right after Office.initialize. 

```
// The initialize function must be run each time a new page is loaded.
Office.initialize = function (reason) {
        
  (window as any).Promise = OfficeExtension.Promise;

  $(document).ready(function () {
```

It would be super helpful if this article was updated to include these two items as they took hours to sort out. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b888399c-fe96-0934-00ce-86db8fa8cf5b
* Version Independent ID: 2c5d24e8-5639-42b1-4d12-119f89591550
* Content: [Convert an Office Add-in project in Visual Studio to TypeScript - Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/develop/convert-javascript-to-typescript)
* Content Source: [docs/develop/convert-javascript-to-typescript.md](https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/develop/convert-javascript-to-typescript.md)
* Product: **non-product-specific**
* Technology: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

### PR DESCRIPTION
**Pull request by kbrandl**
Fixes #1142

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[56d42f1](https://github.com/OfficeDev/office-js-docs-pr/pull/1158/commits/56d42f1ec66f0ecb26c55c09b39453fb5f671d38)_: 

:clock10: **Pending**: waiting for processors ([28](https://ops.microsoft.com/#/statistic) builds ahead of you) 

 :warning: **Docs Build is busy**, currently there are 28 builds ahead of this one, for more information you can view the Build queue graph on the [Docs Portal](https://ops.microsoft.com/#/statistic).

**Comment by o365devx** [comment]
Docs Build status updates of commit _[56d42f1](https://github.com/OfficeDev/office-js-docs-pr/pull/1158/commits/56d42f1ec66f0ecb26c55c09b39453fb5f671d38)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/convert-javascript-to-typescript.md](https://github.com/OfficeDev/office-js-docs-pr/blob/kb-fix-1142/docs/develop/convert-javascript-to-typescript.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C7%5C8%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201907082225233822-1158%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=4V6eszlyoWHHEdmz%2F%2BEhkggd4tu2xdxT5cSoYgUIei8%3D&st=2019-07-08T22%3A21%3A28Z&se=2019-08-08T22%3A26%3A28Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/develop/convert-javascript-to-typescript.md [hunk 1]
```diff
@@ -78,13 +79,19 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
 	declare var fabric: any;
 	```
 
-12. In the **Home.ts** file, change **'1.1'** to **1.1** (that is, remove the quotation marks) in the following line:
+12. In the **Home.ts** file, find the line `Office.initialize = function (reason) {` and add the following line immediately after it, as the first line inside the function:
+
+    ```typescript
+    (window as any).Promise = OfficeExtension.Promise;
+    ```
```
Comment (AlexJerabek) [inline docs/develop/convert-javascript-to-typescript.md]: I think this is fine, but for the sake of discussion, you it make more sense to frame this addition in context? Like,
```typescript
Office.initialize = function (reason) {
    // add the following line
    (window as any).Promise = OfficeExtension.Promise;
…
```

Comment (kbrandl) [reply docs/develop/convert-javascript-to-typescript.md]: Good suggestion, will do.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[44edf62](https://github.com/OfficeDev/office-js-docs-pr/pull/1158/commits/44edf625d8e0675c9835c096ad11d3bdf80c503c)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/convert-javascript-to-typescript.md](https://github.com/OfficeDev/office-js-docs-pr/blob/kb-fix-1142/docs/develop/convert-javascript-to-typescript.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C7%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201907101721560588-1158%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=GfpntVlQkJmwxiECr%2BWgBZa3A7okWoC6TBZxuMc%2BA00%3D&st=2019-07-10T17%3A17%3A52Z&se=2019-08-10T17%3A22%3A52Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[85d8840](https://github.com/OfficeDev/office-js-docs-pr/pull/1158/commits/85d8840df26928f4c4d8c156358f3561e74e046f)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/convert-javascript-to-typescript.md](https://github.com/OfficeDev/office-js-docs-pr/blob/kb-fix-1142/docs/develop/convert-javascript-to-typescript.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C7%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201907101728528370-1158%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=B3XVyISn23wEpWKWrM21ZD0PMa3XNRbB3nY7Fi%2F56Xk%3D&st=2019-07-10T17%3A24%3A55Z&se=2019-08-10T17%3A29%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[4a7e8fd](https://github.com/OfficeDev/office-js-docs-pr/pull/1158/commits/4a7e8fda35284ce86865614812e254f0b0cab34f)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/convert-javascript-to-typescript.md](https://github.com/OfficeDev/office-js-docs-pr/blob/kb-fix-1142/docs/develop/convert-javascript-to-typescript.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C7%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201907101735099116-1158%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=vd2d0i%2F%2FdGn%2FvPo45GR6dVaQLjo2UbwRQS3mrpvhHkg%3D&st=2019-07-10T17%3A31%3A11Z&se=2019-08-10T17%3A36%3A11Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**